File: 2023/recursion/homework/cpp/question01.cpp
// factorial using recursion

#include <iostream>

using namespace std;

int factorial(int);

int main()
{
    int input, output = 0;
    cout << input << " " << output << "\n";
    cout << "Enter the number for which you want to find factorial: ";
    cin >> input;

    output = factorial(input);
    cout << "Factorial of " << input << " is: " << output << "\n";
}

int factorial (int num)
{
    if (num == 1)
    {
        return num;
    }

    return num * factorial(num - 1);
}

File: 2023/recursion/homework/go/question01.go
// factorial using recursion

package main

import "fmt"

func factorial(num int) int {
	if num == 1 {
		return num
	}
	return num * factorial(num-1)
}

func main() {
	var input, output int
	fmt.Print("Enter the number for which you want to find factorial: ")
	fmt.Scan(&input)

	output = factorial(input)
	fmt.Printf("Factorial of %v is %v\n", input, output)
}
